components/core/tests/test-ffi_Value.cpp

```diff
@@ -0,0 +1,195 @@
+
+    vector<string> dict_vars;
+    for (size_t i{0}; i < dict_var_bounds.size(); i += 2) {
+        auto const begin_pos{static_cast<size_t>(dict_var_bounds[i])};
+        auto const end_pos{static_cast<size_t>(dict_var_bounds[i + 1])};
+        dict_vars.emplace_back(text.cbegin() + begin_pos, text.cbegin() + end_pos);
+    }
+
+    return clp::ir::EncodedTextAst<encoded_variable_t>{logtype, dict_vars, encoded_vars};
+}
+
+template <typename Type>
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+auto test_type_probing(Value const& value) -> void {
+    REQUIRE((std::is_same_v<std::monostate, Type> == value.is_null()));
+    REQUIRE((std::is_same_v<value_int_t, Type> == value.is<value_int_t>()));
+    REQUIRE((std::is_same_v<value_float_t, Type> == value.is<value_float_t>()));
+    REQUIRE((std::is_same_v<value_bool_t, Type> == value.is<value_bool_t>()));
+    REQUIRE((std::is_same_v<string, Type> == value.is<string>()));
+    REQUIRE((std::is_same_v<EightByteEncodedTextAst, Type> == value.is<EightByteEncodedTextAst>()));
+    REQUIRE((std::is_same_v<FourByteEncodedTextAst, Type> == value.is<FourByteEncodedTextAst>()));
+}
+
+template <typename Type>
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+auto test_typed_value_probing(Value const& value, Type const& typed_value) -> void {
+    if constexpr (std::is_same_v<value_int_t, Type>) {
+        REQUIRE((value.get_immutable_view<Type>() == typed_value));
+        REQUIRE((std::is_same_v<value_int_t, decltype(value.get_immutable_view<Type>())>));
+    } else {
+        REQUIRE_THROWS(value.get_immutable_view<value_int_t>());
+    }
+
+    if constexpr (std::is_same_v<value_float_t, Type>) {
+        REQUIRE((value.get_immutable_view<Type>() == typed_value));
+        REQUIRE((std::is_same_v<value_float_t, decltype(value.get_immutable_view<Type>())>));
+    } else {
+        REQUIRE_THROWS(value.get_immutable_view<value_float_t>());
+    }
+
+    if constexpr (std::is_same_v<value_bool_t, Type>) {
+        REQUIRE((value.get_immutable_view<Type>() == typed_value));
+        REQUIRE((std::is_same_v<value_bool_t, decltype(value.get_immutable_view<Type>())>));
+    } else {
+        REQUIRE_THROWS(value.get_immutable_view<value_bool_t>());
+    }
+
+    if constexpr (std::is_same_v<string, Type>) {
+        REQUIRE((value.get_immutable_view<Type>() == typed_value));
+        REQUIRE((std::is_same_v<std::string_view, decltype(value.get_immutable_view<Type>())>));
+    } else {
+        REQUIRE_THROWS(value.get_immutable_view<string>());
+    }
+
+    if constexpr (std::is_same_v<EightByteEncodedTextAst, Type>) {
+        REQUIRE((value.get_immutable_view<Type>().get_logtype() == typed_value.get_logtype()));
+        REQUIRE((value.get_immutable_view<Type>().get_dict_vars() == typed_value.get_dict_vars()));
+        REQUIRE(
+                (value.get_immutable_view<Type>().get_encoded_vars()
+                 == typed_value.get_encoded_vars())
+        );
```

Comment:
Instead of this, do you just want to add an equality operator to `EncodedTextAst`? It'd be a one-liner:

```cpp
    auto operator==(EncodedTextAst const&) const -> bool = default;
```